Py3 qt5

Valdas2000 opened the pull request:
Hello.
Seems my changes looks good with your updates. 
There are following fixes introduced:
1 - Fixed issue in open files saved from a different language (custom pattern translation may be lost)
The PNG file saved in other than English localization opens with no errors in English version (cuts fine but fills change to defaults)
The PNG saved in English localization opens fine in any localization 
/I will continue efforts to get complete localization compatibility/

2 - Change labels in Incra Template
Seems the template became more informative now.

3 - Fixed print issue for colored Templates.
The PNG image converted into sRGB profile (the profile saved into the file). So it should be printed well and displayed correct colors  on various monitors

4 - Code refactored. Decimal aliased to D. 
It just to simplify the code,

5 - _future_ removed. Seems we need no future anymore because on python3.
The point is that print is already in python3
Python3 offers float as result of integer division. This code was refactored
lrange migrated to range

Valdas2000 committed:
Merge pull request #2 from Valdas2000/py3_qt5

Py3 qt5

Valdas2000 committed:
Merge pull request #8 from Valdas2000/py3_qt5

Py3 qt5

Valdas2000 committed:
Merge pull request #9 from Valdas2000/py3_qt5

Py3 qt5

Valdas2000 committed:
[~] Refactoring. Updated labels in Incra templates

Valdas2000 committed:
[+] Add ICC profile to PNG, Fix file open

1. The saved png now contain embeded sRGB profile for color much
2. Fixed issue in open files saved from a different language (custom pattern translation may be lost)

Valdas2000 committed:
Merge remote-tracking branch 'upstream/master'

Valdas2000 committed:
Merge branch 'master' into py3_qt5

Valdas2000 committed:
Merge pull request #11 from Valdas2000/py3_qt5

Py3 qt5

lowrie commented:
@Valdas2000 I'm having difficulty with my python installation and
```
from PIL import ImageCms
```
I get
```
Traceback (most recent call last):
  File "/anaconda3/lib/python3.6/site-packages/PIL/ImageCms.py", line 23, in <module>
    from PIL import _imagingcms
ImportError: cannot import name '_imagingcms'
```
This appears to be a common issue with anaconda.  I'll try to figure out a way around it, but it may delay merging this PR.

Valdas2000 commented:
Hello
BTW. I used Pillow 5.0.0 which is a PIL fork. May be that is the cause
pip install Pillow

There was an issue https://github.com/ContinuumIO/anaconda-issues/issues/371 But I did not dig into the fix. (Mac is too complex for me)

Valdas2000 committed:
[~] Change lables on Incra Template

lowrie commented:
Yes, I saw the 371 issue.  For some reason, anaconda is not supporting littlecms in the pillow build.  This issue has been around for many years.

Valdas2000 committed:
[+] New config paramiter Line width

Valdas2000 committed:
[~] Line Width in config file

Valdas2000 committed:
[-] Platformspecific workaround for  ContinuumIO/anaconda-issues#371

Valdas2000 commented:
Got one more question.
Let's say we are going to print a saved PNG the point is we have "fixed" size
So it is not possible to get properly size of print

Do you have any Ideas 

(For now I add dimensions to the lower left corner).

lowrie commented:
This looks good, but I want to change a few minor things before merging.  See my push to this in your repo.

Valdas2000 committed:
[+]Dimentions mark

Valdas2000 commented:
Ok. I will wait for your commits.

Valdas2000 committed:
[+] Ability to set image size in resolution (ppi)

This feature allow to printimagein apropersize with less efforts

Valdas2000 committed:
[+] Fix: Watermark was not saved. So it was lost on image open.

Valdas2000 committed:
[+] Keep the last directory and file name saved.

No need to jump over folders anymore.

Valdas2000 committed:
[~] Open file fix (unable to upen saved files in russion)

The problem was that the description is translated so now we are using part of the class name.

Valdas2000 committed:
[~] Fix dpi issue.

Screen copy - application saves screen resolution
Save file
 - calculated In case image size specifyed in pixels
 - fixed incase image size specifyed in DPI
-------
Now we got functionalitydifference inscreenshot and saved file.
screen is to share expirience
saved file is for printing as well

Valdas2000 committed:
[~]Fix.Conversionrequiredfor comparation

Valdas2000 committed:
[~] Set Depth of cut limitations for dovetails

lowrie commented on config_file.py:
```diff
@@ -223,6 +226,7 @@
 # common default values.  The canvas, board, and template colors from
 # http://paletton.com/#uid=50F0u0kllllaFw0g0qFqFg0-obq
 COMMON_VALS = {'version': 'NONE',
+               'last_path': '\\pyrouterjig.png',
```
On a Mac, this results in the default filename `\pyrouterjig0.png` in the current working directory. Removing the `\\` fixed the issue.

Valdas2000 committed:
[~] Fix the default path in last_path

Valdas2000 commented:
Seems it is also need to keep the path not only on save but on open as well.
May be it will be also good to have Save and Save as... 
It's not a big deal but I am not sure if it is make sense for our tasks.

Valdas2000 committed:
[~] Refactored save config, application title,  documentation... reference

Valdas2000 committed:
ALIGN fixed, russian improvied

russian translation improvied
template allignment issue fixed
(previous ALIGN was placed wrong). Now templetes work with no issue.
